src/map/atcommand.c

```diff
@@ -8998,19 +9032,52 @@ ACMD(addperm)
 	return true;
 }
 
+/**
+ * Unloads a script file.
+ * Note: Be aware that some changes made by NPC are not reverted on unload. See doc/atcommands.txt for details.
+ *
+ * @code{.herc}
+ *	@unloadnpcfile <path> {<flag>}
+ * @endcode
+ *
+ **/
 ACMD(unloadnpcfile)
 {
-	if (!*message) {
-		clif->message(fd, msg_fd(fd,1385)); // Usage: @unloadnpcfile <file name>
+	char file_path[100] = {'\0'};
+	int flag = 1;
+
+	if (*message == '\0' || (sscanf(message, "%99s %1d", file_path, &flag) < 1)) {
+		clif->message(fd, msg_fd(fd, 1385)); /// Usage: @unloadnpcfile <path> {<flag>}
 		return false;
 	}
 
-	if (npc->unloadfile(message)) {
-		clif->message(fd, msg_fd(fd,1386)); // File unloaded. Be aware that mapflags and monsters spawned directly are not removed.
-	} else {
-		clif->message(fd, msg_fd(fd,1387)); // File not found.
+	const char *dot = strrchr(file_path, '.');
+
+	if (dot == NULL) {
+		clif->message(fd, msg_fd(fd, 1518)); /// No file extension specified.
 		return false;
 	}
+
+	if (strlen(dot) < 4 || strncmp(dot, ".txt", 4) != 0) {
```

Comment:
about the `.txt` this checking has forced all users to use `.txt` file extension for all npc scripts?
previously, emulator doesn't have a strict checking for file extension, which end up allow user to use whatever extension they want.
Over the past decade, I often see a lot users use different file extension like `.c` `.abc` `.name` `.whatever` it is...

perhaps remove the `.txt` checking? since it block other users who use random file extension.

Reply:
Really? I never saw someone using custom file extensions for NPCs. :flushed:

Well, I guess checking for the dot is save enough to ensure that a file was passed and not a directory.
I'll changes this later today.

Reply:
Changed! (See `is_file()` function(s) in src/common/utils.c.)